### ISSUE TITLE
 null is not an object (evaluating 'this.a["helper-iframe-url"].indexOf')

### ISSUE THREAD

**Issue by erwinmombay** [ampproject/amphtml#11184]
stack:

installHelperFrame@https://cdn.ampproject.org/rtv/011504040004635/v0/amp-web-push-0.1.js:35:71
start@https://cdn.ampproject.org/rtv/011504040004635/v0/amp-web-push-0.1.js:34:86
buildCallback@https://cdn.ampproject.org/rtv/011504040004635/v0/amp-web-push-0.1.js:24:203
https://cdn.ampproject.org/rtv/011504040004635/v0.js:176:244
initializePromise@[native code]
Promise@[native code]
build@https://cdn.ampproject.org/rtv/011504040004635/v0.js:176:198
build@https://cdn.ampproject.org/rtv/011504040004635/v0.js:157:150
Di@https://cdn.ampproject.org/rtv/011504040004635/v0.js:279:332
Ci@https://cdn.ampproject.org/rtv/011504040004635/v0.js:278:159
upgraded@https://cdn.ampproject.org/rtv/011504040004635/v0.js:280:297
Dc@https://cdn.ampproject.org/rtv/011504040004635/v0.js:175:29
Te@https://cdn.ampproject.org/rtv/011504040004635/v0.js:187:306

sample page it occurred: https://www-babnet-net.cdn.ampproject.org/v/s/www.babnet.net/mobile/cadredetail-amp.php?id=147343&amp_js_v=0.1&usqp=mq331AQCCAE%3D

user agent:
Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/601.1 (KHTML, like Gecko) CriOS/53.0.2785.109 Mobile/14G60 Safari/601.1.46

**Comment by jridgewell** [comment]
Unminified:

```js
null is not an object (evaluating 'this.a["helper-iframe-url"].indexOf')
at indexOf (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/extensions/amp-web-push/0.1/web-push-service.js:208)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/extensions/amp-web-push/0.1/web-push-service.js:158)
at config (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/extensions/amp-web-push/0.1/amp-web-push-config.js:142)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/custom-element.js:778)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/custom-element.js:777)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/service/resource.js:316)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/service/resources-impl.js:585)
at buildResourceUnsafe_ (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/service/resources-impl.js:527)
at buildOrScheduleBuildForResource_ (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/service/resources-impl.js:646)
at (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/custom-element.js:715)
at impl (https://raw.githubusercontent.com/ampproject/amphtml/1504040004635/src/custom-element.js:1090)
```

**Comment by erwinmombay** [comment]
@jasonpang would you be able to fix this?

@aghassemi can you triage this if @jasonpang can't work on it?

appreciate it

**Comment by aghassemi** [comment]
will plan for next sprint if @jasonpang can't get to it.

**Comment by jasonpang** [comment]
We're investigating what caused the issue.

**Comment by jasonpang** [comment]
When amp-web-push starts up, it checks whether the browser environment supports web push notifications. If the environment doesn't support web push notifications, the code incorrectly `return`s ... into the next method and just continues running. Eventually, configuration variables that were supposed to be set in the initialization method we just quit out of, are accessed, but are null, and the whole thing errors. Will quickly fix this to correctly quit out of initializing amp-web-push and not just that one method. Will add a test as well.

**Comment by aghassemi** [comment]
Thanks @jasonpang

**Comment by erwinmombay** [comment]
@jasonpang thanks so much for the quick turnaround! 👏

**Comment by jasonpang** [comment]
@erwinmombay @aghassemi Got a PR up here https://github.com/ampproject/amphtml/pull/11192

**Comment by aghassemi** [comment]
fixed by #11192
